src/Controller/Api/ApiController.php

```diff
@@ -121,21 +95,23 @@ private function getDataFromFile(): array
     /**
      * Returns formatted list of resources.
      *
-     * @param Request $request
      * @param array $data
+     * @param UrlGeneratorInterface $urlGenerator
      *
      * @return array
      */
-    private function prepareData(Request $request, array $data): array
+    private function generateUrls(array $data, UrlGeneratorInterface $urlGenerator): array
```

Comment:
This interface is not clear. Try avoding usage of plain arrays in the code as it's hard to maintein for other developers. An interface must be clear without the need to check the implementation.